lib/kafo/execution_environment.rb

```diff
@@ -0,0 +1,78 @@
+require 'tmpdir'
+
+require 'kafo/fact_writer'
+require 'kafo/hiera_configurer'
+require 'kafo/puppet_configurer'
+
+module Kafo
+  class ExecutionEnvironment
+    def initialize(config, logger = KafoConfigure.logger)
+      @config = config
+      @logger = logger
+    end
+
+    def directory
+      @directory ||= begin
+        directory = Dir.mktmpdir('kafo_installation')
+        @logger.debug("Creating execution environment in #{directory}")
+        directory
+      end
+    end
+
+    def store_answers
+      answer_data = HieraConfigurer.generate_data(@config.modules, @config.app[:order])
+      File.open(answer_file, 'w') { |f| f.write(YAML.dump(answer_data)) }
```

Comment:
how about debug log with information "writing to file #{answer_file}" on line above this one?